### PR TITLE
FE: Topic: Validate ISR/replication upon creation

### LINKED ISSUE #3152: Topic: Validate ISR/replication upon creation
It's currently possible to edit a topic so the min iSR > replication factor.

### PR DESCRIPTION
**Pull request by ShantalaGaitonde**
- [ ] **Breaking change?** (if so, please describe the impact and migration path for existing application instances)

**What changes did you make?** (Give an overview)
Fixes #3152 
- Updated yupExtended.ts to add validation to minInSyncReplicas and replicationFactor
- Updated unit tests

**Is there anything you'd like reviewers to focus on?**
No

**How Has This Been Tested?** (put an "x" (case-sensitive!) next to an item)

- [ ] No need to
- [x] Manually (please, describe, if necessary)
- [x] Unit checks
- [ ] Integration checks
- [ ] Covered by existing automation

**Checklist** (put an "x" (case-sensitive!) next to all the items, otherwise the build will fail)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (e.g. **ENVIRONMENT VARIABLES**)
- [x] My changes generate no new warnings (e.g. Sonar is happy)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged

**A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: kafka-ui-react-app/src/lib/__test__/yupExtended.spec.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-import { isValidEnum, isValidJsonObject } from 'lib/yupExtended';
+import { isValidEnum, isValidJsonObject, topicFormValidationSchema } from 'lib/yupExtended';
```
Comment (Mgrdich) [inline kafka-ui-react-app/src/lib/__test__/yupExtended.spec.ts:1]: please check the linter build and test warnings

File: kafka-ui-react-app/src/lib/yupExtended.ts [hunk 2]
```diff
@@ -105,8 +105,16 @@ export const topicFormValidationSchema = yup.object().shape({
     .max(2147483647)
     .required()
     .typeError('Number of partitions is required and must be a number'),
-  replicationFactor: yup.string(),
-  minInSyncReplicas: yup.string(),
+  replicationFactor: yup
+    .number()
+    .transform((_, val) => (isNaN(val) ? 0 : Number(val)))
```
Comment (Mgrdich) [inline kafka-ui-react-app/src/lib/yupExtended.ts:110]: please check this error.

Unexpected use of 'isNaN'. Use Number.isNaN instead https://github.com/airbnb/javascript#standard-library--isnan

File: kafka-ui-react-app/src/lib/yupExtended.ts [hunk 3]
```diff
@@ -105,8 +105,16 @@ export const topicFormValidationSchema = yup.object().shape({
...
+    .min(yup.ref('minInSyncReplicas'),
+      "Replication Factor must be greater than Min In Sync Replicas"),
+  minInSyncReplicas: yup
+    .number()
+    .transform((_, val) => (isNaN(val) ? 0 : Number(val)))
```
Comment (Mgrdich) [inline kafka-ui-react-app/src/lib/yupExtended.ts:115]: please check this error

Unexpected use of 'isNaN'. Use Number.isNaN instead https://github.com/airbnb/javascript#standard-library--isnan

**Review by Mgrdich — CHANGES_REQUESTED** [review]
Please configure ESLint and prettier with our configs otherwise the pipelines will fail.

eslint will help you to catch alot of the erorrs,

File: kafka-ui-react-app/src/lib/yupExtended.ts (hunk 2 above)
Comment (ShantalaGaitonde) [reply kafka-ui-react-app/src/lib/yupExtended.ts]: No longer a problem.

File: kafka-ui-react-app/src/lib/yupExtended.ts (hunk 3 above)
Comment (ShantalaGaitonde) [reply kafka-ui-react-app/src/lib/yupExtended.ts]: Handled.

File: kafka-ui-react-app/src/lib/__test__/yupExtended.spec.ts (hunk 1 above)
Comment (ShantalaGaitonde) [reply kafka-ui-react-app/src/lib/__test__/yupExtended.spec.ts]: Has been fixed.

File: kafka-ui-react-app/src/lib/yupExtended.ts [hunk 4]
```diff
@@ -90,33 +90,68 @@ export function cacheTest(
...
+        'Only alphanumeric, _, -, and . allowed'
+      ),
+    partitions: yup
+      .number()
+      .min(1)
+      .max(2147483647)
+      .required()
+      .typeError('Number of partitions is required and must be a number'),
+    replicationFactor: yup.lazy((value) => {
+      if (value && value !== '') {
```
Comment (Mgrdich) [inline kafka-ui-react-app/src/lib/yupExtended.ts:110]: isn't the `if (value)` enough ? cause it already checks if it is not `falsy`

File: kafka-ui-react-app/src/lib/yupExtended.ts [hunk 5]
```diff
@@ -90,33 +90,68 @@ export function cacheTest(
...
+                'Replication Factor must be greater than Min In Sync Replicas'
+              )
+              .required(),
+          otherwise: () => yup.number().min(0),
+        });
+      }
+      return yup.string();
+    }),
+    minInSyncReplicas: yup.lazy((value) => {
+      if (value && value !== '') {
```
Comment (Mgrdich) [inline kafka-ui-react-app/src/lib/yupExtended.ts:127]: same.

File: kafka-ui-react-app/src/lib/yupExtended.ts [hunk 6]
```diff
@@ -90,33 +90,68 @@ export function cacheTest(
...
+    partitions: yup
+      .number()
+      .min(1)
+      .max(2147483647)
+      .required()
+      .typeError('Number of partitions is required and must be a number'),
+    replicationFactor: yup.lazy((value) => {
+      if (value && value !== '') {
+        return yup.number().when('minInSyncReplicas', {
+          is: (val) => val && val !== '',
```
Comment (Mgrdich) [inline kafka-ui-react-app/src/lib/yupExtended.ts:112]: same thing here , you can use `!!val` if it has boolean type

**Pull request closed without merging**